docs/4-language-usage/1-general/g-1050.md

```diff
@@ -9,53 +9,55 @@ Literals are often used more than once in your code. Having them defined as a co
 
 All constants should be collated in just one package used as a library. If these constants should be used in SQL too it is good practice to write a deterministic package function for every constant.
 
+In specific situations this rule could lead to an extreme plethora of constants, for example if you use Logger like `logger.append_param(p_params =>l_params, p_name => 'p_param1_todo', p_val => p_param1_todo);`, where the value for `p_name` always should be the name of the variable that is passed to `p_val`. For such cases it would be overkill to add constants for every single variable name you are logging, so if you use Logger or similar, consider making that an exception to the rule, just document exactly which exceptions you will allow and stick to them.
```

Comment:
I typically disable this rule. It's too extreme. Currently the check looks for literals not part of a constant declarations. Exceptions are `0` and `1`.  Adding logger as part of the exception is not really a problem. However, if another custom package is used, then this does not work without external configuration. So, maybe it would simplify things if we change the rule that if a literal is used more than 3 times than a constant must be defined. We should make this threshold configurable. In the end it's a similar check as this Java specific [rule](https://rules.sonarsource.com/java/RSPEC-1192?search=literal). This would reduce the number of "false positives" and make the rule more helpful in real life IMO.

Reply:
I don't like the threshold solution.
You are basically saying that a hard coding is okay when it's only there once.
Okay, I could set it to 1 but then the original problem with logger remains.
I'm for a configuration option for logger or similar things:

Disable checks for lines starting with : "my_logger|my_special_xyz"

Reply:
I see. Actually, if we add a way to configure the validator with a configuration (for the CLI, the SonarQube plug and the SQL Developer extension) then we can configure a threshold and logger package name(s).

In other words. The guideline may stay "as is". Details can be handled by the validation check.

Reply:
I'll leave the rule (with the added explanation). Then we'll see what might be done in validators.